app/controllers/stars_controller.rb

```diff
@@ -1,23 +1,14 @@
 # frozen_string_literal: true
 class StarsController < ApplicationController
-  before_action :require_project
+  include CurrentProject
 
   def create
-    current_user.stars.create!(project: @project)
+    if star = current_user.stars.find_by_project_id(current_project.id)
+      star&.destroy
+    else
+      current_user.stars.create!(project: current_project)
+    end
```

Comment:
Add a comment to the `#create` method saying it's actually doing a create/destroy toggle. Otherwise I'd be pretty confused looking at the implementation.